### DISCUSSION TITLE
Parameter Sensitivity

### DISCUSSION THREAD

**Discussion by goncalosousatorres** [casadi/casadi#3930] [Q&A]
Hi all,
I am starting to use Casadi for a project which will involve optimal control and general optimization. I am still learning the framework, and for now I am testing it basic functions within a forward simulation. When I run the simulation with standard Matlab variables, I get the solution quite fast (within a few seconds). When I instead convert one of the model parameters to Casadi SX object, the integration takes much longer (minutes). Is this a normal behavior?

Thank you in advance!

**Comment by jaeandersson** [comment]
What is it that takes time in CasADi? Constructing the dynamic system, creating an integrator, simulating?
